src/pkg/runner/actions_test.go

```diff
@@ -0,0 +1,454 @@
+			wantErr: false,
+		},
+		{
+			name: "Invalid wait action",
+			args: args{
+				wait:    zarfTypes.ZarfComponentActionWait{},
+				timeout: IntPtr(30),
+			},
+			want:    "",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := convertWaitToCmd(tt.args.wait, tt.args.timeout)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("convertWaitToCmd() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("convertWaitToCmd() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func IntPtr(i int) *int {
+	return &i
+}
+
+func Test_validateActionableTaskCall(t *testing.T) {
+	type args struct {
+		inputTaskName string
+		inputs        map[string]types.InputParameter
+		withs         map[string]string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Valid task call with all required inputs",
+			args: args{
+				inputTaskName: "testTask",
+				inputs: map[string]types.InputParameter{
+					"input1": {Required: true, Default: ""},
+					"input2": {Required: true, Default: ""},
+				},
+				withs: map[string]string{
+					"input1": "value1",
+					"input2": "value2",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Invalid task call with missing required input",
+			args: args{
+				inputTaskName: "testTask",
+				inputs: map[string]types.InputParameter{
+					"input1": {Required: true, Default: ""},
+					"input2": {Required: true, Default: ""},
+				},
+				withs: map[string]string{
+					"input1": "value1",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Valid task call with default value for missing input",
+			args: args{
+				inputTaskName: "testTask",
+				inputs: map[string]types.InputParameter{
+					"input1": {Required: true, Default: "defaultValue"},
+					"input2": {Required: true, Default: ""},
+				},
+				withs: map[string]string{
+					"input2": "value2",
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := validateActionableTaskCall(tt.args.inputTaskName, tt.args.inputs, tt.args.withs); (err != nil) != tt.wantErr {
+				t.Errorf("validateActionableTaskCall() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestRunner_performAction(t *testing.T) {
+	type fields struct {
+		TemplateMap map[string]*zarfUtils.TextTemplate
+		TasksFile   types.TasksFile
+		TaskNameMap map[string]bool
+		envFilePath string
+	}
+	type args struct {
+		task   types.Task
+		action types.Action
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		//TODO: Add more test cases
+		{
+			name: "failed action processing due to invalid command",
+			fields: fields{
+				TemplateMap: make(map[string]*zarfUtils.TextTemplate),
+				TasksFile:   types.TasksFile{},
+				TaskNameMap: make(map[string]bool),
+				envFilePath: "test/path",
```

Comment:
I think you want to set the envFilePath to "" for this test case. Currently both test cases are failing with the same error unable to find this envFile.

Reply:
Thanks, Copy paste bug ;).